src/history/api.py

```diff
@@ -397,3 +398,45 @@ def pdf_exists(year: str | int,
         bool: True if and only if the PDF file exists.
     """
     return Path.is_file(get_pdf_filepath(year, semester, student_type, round_num))
+
+
+def get_latest_year_and_sem_with_coursereg_info() -> tuple[str, str]:
+    """
+    Returns the latest year/sem that has coursereg PDF data.
+
+    Returns:
+    ----
+        tuple[str, str]: Tuple containing (acad year, sem).
+    """
+
+    curYear = datetime.now().year
+
+    def get_acad_year_starting_this_calendar_year(curYear: int) -> str:
+        """
+        Returns the later AY starting in curYear.
+
+        Returns:
+        ---
+            str: Later AY starting in current year.
+        """
+        lastTwoDigits = curYear % 100
+        return str(lastTwoDigits) + str(lastTwoDigits + 1)
```

Comment:
Do you think that ensuring the case of lastTwoDigits overflowing from `99` to `100` and `02` being simplified to `2` is a worthwhile corner case to fix? The repository surely will not last that long - but if there is an elegant fix you can think of it would not do much harm to add it.